src/vmm/src/vmm_config/vsock.rs

```diff
@@ -97,51 +97,29 @@ pub(crate) mod tests {
     use super::*;
     use utils::tempfile::TempFile;
 
-    // Placeholder for the path where a socket file will be created.
-    // The socket file will be removed when the scope ends.
-    #[derive(Clone)]
-    pub(crate) struct TempSockFile {
-        path: String,
-    }
-
-    impl TempSockFile {
-        pub fn new(tmp_file: TempFile) -> Self {
-            TempSockFile {
-                path: String::from(tmp_file.as_path().to_str().unwrap()),
-            }
-        }
-        pub fn path(&self) -> &String {
-            &self.path
-        }
-    }
-
-    impl Drop for TempSockFile {
-        fn drop(&mut self) {
-            let _ = std::fs::remove_file(&self.path);
-        }
-    }
+    pub(crate) fn default_config() -> VsockDeviceConfig {
+        let mut tmp_sock_file = TempFile::new().unwrap();
```

Comment:
This is no good. With this code, the tmp-file will delete itself at the end of **this** helper function (`default_config()`).
We actually want to delete itself at the end of the test _that is using_ this helper function.

With this code, tests don't delete their created resources at the temporary paths generated by `TempFile`.

Reply:
Thank you for the comment. Updated `default_config` to take `&TempFile` as parameter to prevent `TempFile` to be dropped in the function.